content/blog/2022/2022-07-03-getting-started-with-aws-step-functions.md

```diff
@@ -0,0 +1,549 @@
+---
+authors: [pratikdas]
+title: "Getting Started with Amazon Step Functions"
+categories: ["AWS"]
+date: 2022-02-10 00:00:00 +1100
+excerpt: "AWS Step Functions is a serverless orchestration service by which we can combine AWS Lambda functions and other AWS services to build complex business applications. We can author the orchestration logic in a declarative style using a JSON-based format called the Amazon States Language(ASL). AWS Step functions also provide a powerful graphical console where we can visualize our application’s workflow as a series of steps. In this article, we will introduce the concepts of AWS Step Functions and understand its working with the help of an example."
+image: images/stock/0117-queue-1200x628-branded.jpg
+url: getting-started-with-aws-step-functions
+---
+
+AWS Step Functions is a serverless orchestration service by which we can combine AWS Lambda functions and other AWS services to build complex business applications. We can author the orchestration logic in a declarative style using a JSON-based format called the Amazon States Language(ASL). AWS Step functions also provide a powerful graphical console where we can visualize our application’s workflow as a series of steps.
+
+In this article, we will introduce the concepts of AWS Step Functions and understand its working with the help of an example.
+
+## Step Functions: Basic Concepts
+Let us first understand some basic concepts of Step Functions. 
+### State Machine, State, and Transitions
+A state machine is a mathematical model of computation consisting of different states connected with transitions. AWS Step functions also implement a State Machine to represent the orchestration logic. Each step of the orchestration is represented by a state in the state machine and connected to one or more states through transitions. 
+
+They are represented by a diagram to visualize the current state of a system as shown here:
+
+{{% image alt="State Machine" src="images/posts/aws-step-function/sm.png" %}}
+ State machines have a start and end and contain at least one state. Transitions represent different events that allow the system to transition from one state to another state.
+
+### Amazon State Language (ASL)
+We define a State machine in JSON format in a structure known as the Amazon States Language (ASL) as shown below:
+
+```json
+{
+  "StartAt": "state1",
+  "States": {
+    "state1": {...},
+    "state2": {...},
+    "state3": {...}
+  }
+}
+```
+This structure represents a collection of `3` state objects with the names: `state1`, `state2`, and `state3`. The state machine starts execution from the state named `state1`. 
```

Comment:
Should it be called `step1` to start execution or is it always the first element in the list?